spec/decanter/decanter_extensions_spec.rb

```diff
@@ -3,13 +3,66 @@
 describe Decanter::Extensions do
 
   describe '#decant' do
+    let(:options) { { } }
+    let(:dummy_class) { Class.new { include Decanter::Extensions } }
 
-    let(:args) { { } }
+    context 'when args is a single resource' do
+      let(:args) { { } }
+
+      before(:each) do
+        allow(dummy_class).to receive(:decant_args).and_return(true)
+      end
+
+      it 'calls decant_args with the args and options' do
+        dummy_class.decant(args, options)
+        expect(dummy_class)
+          .to have_received(:decant_args)
+          .with(args, options)
+      end
+    end
+
+    context 'when args is a collection' do
+      let(:args) { [{}, {}] }
+
+      before(:each) do
+        allow(dummy_class).to receive(:decant_collection).and_return(true)
+      end
+
+      it 'calls decant_collection with the args and options' do
+        dummy_class.decant(args, options)
+        expect(dummy_class)
+          .to have_received(:decant_collection)
+          .with(args, options)
+      end
+    end
+  end
+
+  describe '#decant_collection' do
+    let!(:args) { [{ foo: 'bar' }, { foo: 'baz' }] }
+    let(:options) { { } }
+    let(:dummy_class) { Class.new { include Decanter::Extensions } }
+
+    before(:each) do
+      allow(dummy_class).to receive(:decant_args).and_return(true)
+    end
+
+    it 'calls decant_args on each element in the collection' do
+      # expect(dummy_class).to receive(:decant_args).twice
```

Comment:
@chawes13  can either have this commented line and the next line be the spec, or leave what's uncommented. Which do you prefer as a representation of this spec?

Reply:
I prefer what's left uncommented. You can remove this line!